app.js

```diff
@@ -19,12 +20,13 @@ app.get("/api/articles/:article_id", getArticleById);
 
 app.get("/api/articles", getArticles);
 
+app.post("/api/articles/:article_id/comments", postComment);
 app.get("/api/articles/:article_id/comments", getCommentsById);
 
 app.patch("/api/articles/:article_id", patchArticleById);
 
 app.use((err, req, res, next) => {
-  if (err.code === "22P02") {
+  if (err.code === "22P02" || err.code === "23502") {
     res.status(400).send({ msg: "Bad request" });
   } else if (err.status && err.msg) {
```

Comment:
my only beef is that this error handler is send back a message, and you haven't designed this behaviour in the tests.